src/plugins/share/server/url_service/http/short_urls/register_create_route.ts

```diff
@@ -48,19 +49,34 @@ export const registerCreateRoute = (router: IRouter, url: ServerUrlService) => {
         });
       }
 
-      const shortUrl = await shortUrls.create({
-        locator,
-        params,
-        slug,
-        humanReadableSlug,
-      });
+      try {
+        const shortUrl = await shortUrls.create({
+          locator,
+          params,
+          slug,
+          humanReadableSlug,
+        });
 
-      return res.ok({
-        headers: {
-          'content-type': 'application/json',
-        },
-        body: shortUrl.data,
-      });
+        return res.ok({
+          headers: {
+            'content-type': 'application/json',
+          },
+          body: shortUrl.data,
+        });
+      } catch (error) {
+        if (error instanceof UrlServiceError) {
+          if (error.code === 'SLUG_EXISTS') {
+            return res.customError({
+              statusCode: 409,
+              headers: {
+                'content-type': 'application/json',
```

Comment:
Don't think this is needed. Didn't see it used in other usages of `res.customError`